packages/medusa/src/services/product.js

```diff
@@ -185,6 +185,12 @@ class ProductService extends BaseService {
       )
     }
 
+    if (product.variants) {
+      product.variants.sort(
+        (variant1, variant2) => variant1.variant_rank - variant2.variant_rank
+      )
+    }
+
```

Comment:
Can we do the sorting at the database level instead? Potentially in `findOneWithRelations`